pkg/controller/configuration/config_utils.go

```diff
@@ -236,20 +236,33 @@ func buildConfigManagerParams(cli client.Client, ctx context.Context, cluster *a
 
 func GetConfigManagerGRPCPort(containers []corev1.Container) (int32, error) {
 	for _, container := range containers {
-		if found := foundPortByConfigManagerPortName(container); found != nil {
-			return found.ContainerPort, nil
+		if container.Name != constant.ConfigSidecarName {
+			continue
+		}
+		if port, ok := findPortByPortName(container); ok {
+			return port, nil
 		}
 	}
-	return -1, core.MakeError("failed to find config manager grpc port, please add named config-manager port")
+	return constant.InvalidContainerPort, core.MakeError("failed to find config manager grpc port, please add named config-manager port")
 }
 
-func foundPortByConfigManagerPortName(container corev1.Container) *corev1.ContainerPort {
+func allocConfigManagerHostPort(comp *component.SynthesizedComponent) (int32, error) {
+	pm := intctrlutil.GetPortManager()
+	portKey := intctrlutil.BuildHostPortName(comp.ClusterName, comp.Name, constant.ConfigSidecarName, constant.ConfigManagerPortName)
+	port, err := pm.AllocatePort(portKey)
```

Comment:
This port is unique across the Kubernetes cluster, which also means it is a scarce resource that can be easily exhausted. Do we really need to allocate such a port for the config-manager?

Additionally, how are the HTTPS ports for Lorry allocated when using hostNetwork? @xuriwuyun